draft-ietf-quic-http.md

```diff
@@ -859,9 +863,8 @@ expectation of the matching PUSH_PROMISE. The client can use stream flow control
 (see section 4.1 of {{QUIC-TRANSPORT}}) to limit the amount of data a server may
 commit to the pushed stream.
 
-If a promised server push is not needed by the client, the client SHOULD send a
-CANCEL_PUSH frame. If the push stream is already open or opens after sending the
-CANCEL_PUSH frame, the client can abort reading the stream with an error code of
+Push stream data can also arrive after a client has canceled a push. In this
+case, the client can abort reading the stream with an error code of
```

Comment:
Did we lose the `SHOULD send a CANCEL_PUSH`? Does that matter?

Reply:
It moved up to 827, though I demoted it from a SHOULD to a statement about how the frame is used.  If someone cares about the difference, I can re-RECOMMEND it.

Reply:
I'm indifferent

Reply:
Is this change still editorial if normative language is changed?  I don't feel strongly otherwise: I think the SHOULD was justified, but the current test flows better without it given that the following sentence does not have SHOULD either.

Reply:
It has to be design if it prohibits something previously permitted or permits something previously prohibited.  A changed recommendation is hazier, and we've sometimes chosen to treat those as design for better visibility to make sure the WG is fully on-board, but they don't have to be.

Reply:
Okay, that makes sense.  I'm fine with this change then.  Thanks for the explanation.

Reply:
Actually, it turns out the same SHOULD is present in the CANCEL_PUSH section anyway:

> When a server sends CANCEL_PUSH, it is indicating that it will not be fulfilling
> a promise.  The client cannot expect the corresponding promise to be fulfilled,
> unless it has already received and processed the promised response. A server
> SHOULD send a CANCEL_PUSH even if it has opened the corresponding stream.